tools/rewriter/canonicalize_compile_command_paths.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+
+# /// script
+# requires-python = ">=3.4"
+# dependencies = [
+#     "pathlib",
+# ]
+# ///
+
+import json
+from pathlib import Path
+
+def main():
+    """
+    Assuming we're in a build directory containing a `compile_commands.json`,
+    canonicalize (including making absolute) all paths in every compile command,
+    including as part of arguments (detect as a best effort).
+
+    libclangTooling is finicky with relative paths,
+    and it and other tools don't always handle non-canonical paths well,
+    so making all paths in a `compile_commands.json` canonical is helpful.
+    """
+
+    cc_db = Path("compile_commands.json")
+    cc_text = cc_db.read_text()
+    cmds = json.loads(cc_text)
+    for cmd in cmds:
+        dir = Path(cmd["directory"]).resolve()
+        cmd["directory"] = str(dir)
+        
+        def convert_path(path: str) -> str:
+            path: Path = Path(path)
+            if not path.is_absolute():
+                path = dir / path
+            path = path.resolve()
+            return str(path)
+
+        cmd["file"] = convert_path(cmd["file"])
+        if "output" in cmd:
+            cmd["output"] = convert_path(cmd["output"])
+        
+        def try_convert_arg(arg: str) -> str:
+            for prefix in ["", "-I"]:
```

Comment:
Don't you want this order reversed? All strings start with an empty prefix.

Reply:
Is the order important here?  I'm just checking if the whole argument is a path.  And then checking if, minus a `-I` prefix, the rest of the argument is a path.

Reply:
Checking the other direction is half the work (assuming the `-I` matches). It doesn't matter much.

Reply:
Yeah, but we don't know which will match.  Performance doesn't really matter here, and it's Python anyway, so as long as it's correct, I'll just keep it as is.